Add fontFamily fallback support when using declarative @font-face fonts (#236)

jh3y opened the pull request:
- Attempts to resolve #236 by catching if a `fontFamily` declaration is an `Array` and processing each entry.
- For example, `fontFamily: [myFontFace, 'cursive']` => `.ae { font-family: ae, cursive }`
- Intercepts within `renderFontFace` to produce a value string from the `Array` 👍

tajo commented:
This needs some tests too! Something similar to [this](https://github.com/styletron/styletron/blob/master/packages/styletron-engine-atomic/src/client/__tests__/client.browser.js#L120)

chasestarr commented on packages/styletron-engine-atomic/src/client/client.js:
```diff
@@ -199,8 +199,22 @@ class StyletronClient implements StandardEngine {
-    const key = declarationsToBlock(fontFace);
-    return this.fontFaceCache.addValue(key, fontFace);
+    let result = "";
+    if (Array.isArray(fontFace)) {
```
based on the `FontFaceObject` type, it is always an [object](https://github.com/styletron/styletron/blob/master/packages/styletron-standard/src/style-types.js#L765). would be best to properly type the expected incoming values

jh3y replied:
I've updated the param type to allow `Array` of `FontFaceObject` and `string`.

rtsao commented on packages/styletron-engine-atomic/src/client/client.js:
```diff
@@ -198,9 +198,25 @@ class StyletronClient implements StandardEngine {
+    fontFace: FontFaceObject | Array<FontFaceObject | string>,
+  ): string {
+    let result = "";
+    if (Array.isArray(fontFace)) {
```
See the above comment, I think this logic instead should go in `renderDeclarativeRules`: https://github.com/styletron/styletron/blob/e1e1ebd8489e0ecea11783dc35f8170a3f8a3f07/packages/styletron-standard/src/index.js#L50-L52

jh3y commented:
Hey @rtsao 👋 
I moved the logic into `renderDeclarativeRules` and changed the type declaration.
How would the tests look for this? Would `StyletronClient` need to be mocked or imported? There aren't currently any tests in `styletron-standard`. I'd imagine there being a test for `renderDeclarativeRules`?

rtsao commented:
I suppose an integration test with the atomic engine and styletron-standard might be the best thing here. This doesn’t quite exist yet but would probably be similar to the existing hydration tests

jh3y commented:
I've added a basic `driver` test within `styletron-standard` package.

Tests `getCss` output against expected `String` after invoking the `driver` function 👍
